### DISCUSSION TITLE
Problem with installing quarto-cli with the help of rye package manager

### DISCUSSION THREAD

**Discussion by mrossde** [quarto-dev/quarto-cli#10993] [Q&A]
### Description

Dear Community, 

im using quarto into VS-Code some time now and wanted to put the quarto rendering process into a venv. acutally it seems like quarto is rendering in the "quarto_dev" environment which i can access via powershell and conda.
I decided to use rye package manager for the complete project and the quarto render command actually happened outside the .venv

Now i wanted to install quarto-cli with `rye add quarto-cli` with the follow result. 

```
rye add quarto-cli
Added quarto-cli>=1.5.57 as regular dependency
Reusing already existing virtualenv
Generating production lockfile: \\?\D:\_sync\005_HS\proj\Techn_DE\lect-TDE\requirements.lock
Generating dev lockfile: \\?\D:\_sync\005_HS\proj\Techn_DE\lect-TDE\requirements-dev.lock
Installing dependencies
Resolved 171 packages in 92ms
Building quarto-cli==1.5.57
   Built example-project-with-rye @ file:///D:/_sync/005_HS/proj/Techn_DE/lect-TDE
⠴ Preparing packages... (1/2)    
```                                                     

then comes:

```
Built example-project-with-rye @ file:///D:/_sync/005_HS/proj/Techn_DE/lect-TDE                                                                                                         error: Failed to prepare distributions
  Caused by: Failed to fetch wheel: quarto-cli==1.5.57
  Caused by: Build backend failed to build wheel through `build_wheel()` with exit code: 1
--- stdout:
Current working directory: C:\Users\Mario\AppData\Local\uv\built-wheels-v3\index\b2a7eb67d4c26b82\quarto-cli\1.5.57\eP8UWLi1CVgByOkNlM2E7\quarto_cli-1.5.57.tar.gz
running bdist_wheel
running build
running build_py
Downloading and installing quarto-cli binaries...
Downloading https://github.com/quarto-dev/quarto-cli/releases/download/v1.5.57/quarto-1.5.57-win.zip
--- stderr:
error: [Errno 2] No such file or directory: 'quarto_cli\\quarto-1.5.57\\bin\\vendor\\deno-land\\x\\puppeteer@9-0-2\\vendor\\puppeteer-core\\puppeteer\\common\\AriaQueryHandler.d.ts'      
---
error: Installation of dependencies failed in venv at \\?\D:\_sync\005_HS\proj\Techn_DE\lect-TDE\.venv. uv exited with status: exit code: 2
```

my rye version shows after doing `rye self update`
```
0.41.0
```

  
I tried this with severeal versions - same result. I also moved my repo folder to another place with shorter path, but it didnt helped. Do you have an idea why rye is not installing it? 

thanks in advance!
Mario

**Comment by cderv** [comment]
I don't know specifically about **rye**. I suppose you are referring to https://rye.astral.sh/ 

`pip install quarto-cli` works correctly to install https://pypi.org/project/quarto-cli/ (repo is there : https://github.com/quarto-dev/quarto-cli-pypi). So it is possibly something with **rye** that may not be supported. 
````
> pip install quarto-cli
(....)

Building wheels for collected packages: quarto-cli
  Building wheel for quarto-cli (pyproject.toml) ... done
  Created wheel for quarto-cli: filename=quarto_cli-1.5.57-py3-none-any.whl size=119609848 sha256=a1c5abd493d05abd42d27a971a83a65797e58983ab6a92bfe463aac3bda09920
  Stored in directory: c:\users\chris\appdata\local\pip\cache\wheels\43\b7\b1\ddc0a1111209e1258d11911f76c67b71f9d931e6fdfbf34c39

(....)
````
You should ask to **rye** community and then report back here if something needs to be adapted. Or you could advice use on what to do. As I said repo is there: https://github.com/quarto-dev/quarto-cli-pypi - we don't know **rye** specifically and need help to fix if there is a problem on our package side. 

Thanks!

**Comment by mrossde** [comment]
Hello Chris, 
ok thanks for information. I will ask in the rye forum and come back to that if there are news! Greetings, Mario
